packages/app/components/Queue/TA/TAQueueList.tsx

```diff
@@ -130,13 +131,23 @@ export default function TAQueueList({
     question: Question,
     status: QuestionStatus
   ) => {
+    let newQuestions;
+    if (status in ClosedQuestionStatus) {
+      newQuestions = questions?.filter(q => q.id !== question.id);
+    } else if (status === OpenQuestionStatus.Helping) {
+      newQuestions = questions?.map((q) =>
+        q.id === question.id ? { ...q, status, taHelped: { id: user.id } } : q
```

Comment:
Updating the `taHelped` attribute with an object that contains the id allows us to mutate the state so that we can correctly find the `helpingQuestion` (line 103).